### PR TITLE
[+1 MRG]look in ~/.config/scrapy.cfg for user config

### PR DESCRIPTION
**Pull request by nyov**
Just a slight enhancement.
I like to keep my `~` cleaner by having config files in `~/.config/`.

### CODE REVIEW

**Comment by nramirezuy** [comment]
@nyov making a link from the project doesn't work ?

**Comment by nyov** [comment]
~/.config/ is a relatively standard path, like ~/.local/. It usually exists.
If I made a link (I assume you mean a symlink), then that would be counter to having a cleaner `~/` - there'd still be a file there.

**Comment by nramirezuy** [comment]
I meant a symlink inside the project not on the home. But I didn't know it was standard, do you have some source?

Either way I would like to see an implementation using a environment variable.

**Comment by nyov** [comment]
Here's the standard: http://standards.freedesktop.org/basedir-spec/basedir-spec-latest.html#variables
These locations conform with `$XDG_DATA_HOME` and `$XDG_CONFIG_HOME`.

> If $XDG_CONFIG_HOME is either not set or empty, a default equal to $HOME/.config should be used.

So according to this spec, I should check the `$XDG_CONFIG_HOME` env var, and only use the fixed `~/.config/scrapy.cfg`-path if it's not set.
I'll add this in a bit.

File: scrapy/utils/conf.py [hunk 1]
```diff
@@ -5,6 +5,11 @@
 import six
 from six.moves.configparser import SafeConfigParser
 
+try:
+    from xdg.BaseDirectory import xdg_config_home
```
Comment (kmike) [inline scrapy/utils/conf.py]: what's xdg?

**Comment by nyov** [comment]
xdg is "pyxdg" http://freedesktop.org/wiki/Software/pyxdg/
or known as "python-xdg" in debian/ubuntu: https://packages.debian.org/jessie/python-xdg

**Comment by kmike** [comment]
@nyov I think we can't use pyxdg because it is GPL and doesn't support Python 3.

**Comment by nyov** [comment]
Should I rather not depend on that package? The functionality is defined as this and not very likely to change:
```py
_home = os.path.expanduser('~')
xdg_config_home = os.environ.get('XDG_CONFIG_HOME') or os.path.join(_home, '.config')
```

**Comment by nyov** [comment]
Would this work now?

**Comment by kmike** [comment]
LGTM. I personally don't care about this feature, but see zero reasons not to merge it. If it makes life easier for people why not do it.

By the way, this scrapy.cfg lookup chain is not documented anywhere :) Docs only say about scrapy.cfg in the project root.

**Comment by nyov** [comment]
I would add some docs for it, but I can't find a good category?

**Comment by kmike** [comment]
There are some docs for scrapy.cfg in commands.rst ("Default structure of Scrapy projects"); adding a note there would be nice.

**Comment by nyov** [comment]
Alright, I decided to write it under a new heading, hope that works.

**Comment by kmike** [comment]
Thanks @nyov! 

* What about mentioning that the settings from all available scrapy.cfg files are merged?
* Regarding env variables - "Scrapy also understands, and can be configured through, a number of environment variables." doesn't help users because it is not documented which variables Scrapy understands. I see the code which allows to override settings with SCRAPY_... env variables is marked for deprecation/removal. The only relevant env variables seems to be SCRAPY_SETTINGS_MODULE and SCRAPY_PROJECT. It could be better to document them explicitly. 

@nyov @nramirezuy what do you think? I can merge the changes and tweak the docs if this sounds ok (updated PR is also welcome, of course :).

**Comment by nramirezuy** [comment]
:+1:

**Comment by nyov** [comment]
@kmike, I was afraid of that when I rwrote the sentence ;) Okay I will list the ENV vars you mentioned, as I'm not sure what all `SCRAPY_...` would include.

**Comment by kmike** [comment]
Thanks!

**Pull request merged**